internal/resource-request-operator/resourceRequest_controller_test.go

```diff
@@ -0,0 +1,82 @@
+package resourceRequestOperator
+
+import (
+	"context"
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	sharingv1alpha1 "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	"github.com/liqotech/liqo/pkg/discovery"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"time"
+)
+
+var _ = Describe("ResourceRequest controller", func() {
+
+	const (
+		ResourceRequestName = "test-resource"
+		OfferPrefix         = "resourceoffer-"
+		ResourcesNamespace  = "default"
+		timeout             = time.Second * 10
+		interval            = time.Millisecond * 250
+	)
+
+	Context("Testing ResourceRequest Controller when creating a new ResourceRequest", func() {
+		It("Should create new ResourceRequest and related ResourceOffer ", func() {
+			By("By creating a new ResourceRequest")
+			ctx := context.Background()
+			resourceRequest := &discoveryv1alpha1.ResourceRequest{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "discovery.liqo.io/v1alpha1",
+					Kind:       "ResourceRequest",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ResourceRequestName,
+					Namespace: ResourcesNamespace,
+				},
+				Spec: discoveryv1alpha1.ResourceRequestSpec{
+					AuthUrl: "https://127.0.0.1:39087",
+					ClusterIdentity: discoveryv1alpha1.ClusterIdentity{
+						ClusterID: "2468825c-0f62-44d7-bed1-9a7bc331c0b0",
+					},
+					Namespace: ResourcesNamespace,
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, resourceRequest)).Should(Succeed())
+
+			requestLookupKey := types.NamespacedName{Name: ResourceRequestName, Namespace: ResourcesNamespace}
+			createdResourceRequest := &discoveryv1alpha1.ResourceRequest{}
+
+			// We'll need to retry getting this newly created ResourceRequest, given that creation may not immediately happen.
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, requestLookupKey, createdResourceRequest)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
```

Comment:
Why are you waiting for the resource to exist? If the creation at line 47 succeeded it should be there, unless the kubernetes control plane is broken